lms/djangoapps/course_home_api/outline/v1/tests/test_views.py

```diff
@@ -237,6 +238,33 @@ def test_post_course_goal(self):
         assert selected_goal is not None
         assert selected_goal['key'] == 'certify'
 
+    @override_waffle_flag(ENABLE_COURSE_GOALS, active=True)
+    @override_waffle_flag(COURSE_GOALS_NUMBER_OF_DAYS_GOALS, active=True)
+    def test_post_course_goal(self):
+        """ Test that the api returns the correct response when saving a goal """
+        CourseEnrollment.enroll(self.user, self.course.id, CourseMode.AUDIT)
+
+        post_data = {
+            'course_id': self.course.id,
+            'number_of_days_with_visits_per_week_goal': 1,
+            'subscribed_to_goal_reminders': True,
+        }
+        post_course_goal_response = self.client.post(reverse('course-home-save-course-goal'), post_data)
+        assert post_course_goal_response.status_code == 200
+
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+
+        course_goals = response.json()['course_goals']
+        expected_course_goals = {
```

Comment:
this is the format the api currently returns (so the front-end could check whether selected_goal/goal_options have the options for the new or old goals